### PR TITLE
Add validation mode support

### PR DESCRIPTION
**Pull request by henryaddison**
Final part of https://github.com/openactive/data-models/issues/41

Adds a drop-down for selecting validation mode.

Screenshot 2019-11-04 at 16 02 03

Samples drop-down now supports groups depending on usage type (e.g. RDPE or Booking) with optionally supplied associated validation mode for an example (selecting a sample with a validation mode associated with set the site's validation mode to it). The examples in each group are displayed via sub-menus to ensure the initial drop-down is now illegibly long.

Screenshot 2019-11-04 at 16 02 00

### CODE REVIEW

File: src/client/helpers/api-helper.js [hunk 1]
```diff
@@ -31,31 +31,27 @@ export default class ApiHelper {
...
+
+  static validateURL(url, version, validationMode) {
+    return this.validate({ url }, version, validationMode)
+      .then(res => ApiHelper.handleResponse(res));
   }
 
-  static validateURL(url, version) {
-    return fetch(`/api/validateUrl/${version}`, {
+  static validate(data, version, validationMode) {
+    let body = JSON.parse(JSON.stringify(data));
```
Comment (nickevansuk) [inline src/client/helpers/api-helper.js]: This seems fishy?

Comment (henryaddison) [reply src/client/helpers/api-helper.js]: This is something I was going to pick up with Joe during review, for some reason it's not liking the spread operator as I was hoping to just to `const body = { ...date, validationMode };` rather than clone the data object and add validationMode to it.

Comment (benjaminreid) [reply src/client/helpers/api-helper.js]: If the spread isn’t working for whatever reason (weird it doesn’t) I noticed jQuery is in the project, it has it’s own method for that kind of thing https://api.jquery.com/jquery.extend/

TL;DR

```js
const body = $.extend({}, data, { validationMode });
```

Comment (henryaddison) [reply src/client/helpers/api-helper.js]: Thanks, Ben. Just seen this - we went for `Object.assign` which looks like it's the same thing.

Comment (benjaminreid) [reply src/client/helpers/api-helper.js]: Nice! As long as that’s okay with browser support then that’s nice.

Basically no IE and in built Android browsers.

https://caniuse.com/#feat=mdn-javascript_builtins_object_assign

Comment (henryaddison) [reply src/client/helpers/api-helper.js]: Thanks for the warning, Ben. The site is a tool for devs so IE unlikely to be used.
